submissions/Ant-C-tech/memory-pair-game/js/app.js

```diff
@@ -0,0 +1,278 @@
+    congratulationSubText.classList.add('congratulationSubText')
+
+    const newGameBtn = document.createElement('h3')
+    newGameBtn.textContent = 'A samurai has no goal, only a path...'
+    newGameBtn.classList.add('newGame')
+
+    congratulationWrapper.append(congratulationIco, congratulation, congratulationText, congratulationSubText, newGameBtn)
+
+    newGameBtn.addEventListener('click', function() {
+        TASK = []
+        attempts = 0
+        setTask()
+        window.addEventListener('resize', phoneAdaptation)
+        phoneAdaptation()
+        changeContent(createGameField())
+    }, { once: true })
+
+    return congratulationWrapper
+}
+
+function changeContent(content, callback) {
+    MAIN.classList.add('main-hide')
+    MAIN.addEventListener('transitionend', function() {
+        MAIN.innerHTML = ''
+        MAIN.append(content)
+        if (callback) {
+            callback()
+        }
+        MAIN.classList.remove('main-hide')
+    }, { once: true })
+}
+
+function showElem(elem, effect, delay) {
+    if (delay) {
+        elem.classList.add(ANIMATE_DELAY)
+    } else {
+        elem.classList.remove(ANIMATE_DELAY)
+    }
+    elem.style.display = 'block'
+    elem.classList.add(effect)
+    elem.addEventListener('animationend', function() {
+        elem.classList.remove(effect)
+    }, { once: true })
+}
+
+function hideElem(elem, effect, delay) {
+    if (delay) {
+        elem.classList.add(ANIMATE_DELAY)
+    } else {
+        elem.classList.remove(ANIMATE_DELAY)
+    }
+    elem.classList.add(effect)
+    elem.addEventListener('animationend', function() {
+        elem.style.display = 'none'
+        elem.classList.remove(effect)
+    }, { once: true })
+}
+
+function checkAnswer() {
+    MAIN.classList.add('main-notActive')
+    const openCards = document.querySelectorAll('.card-rotate')
+    if (openCards.length === 2) {
+        if (openCards[0].getAttribute('data-value') === openCards[1].getAttribute('data-value')) {
+            for (const card of openCards) {
+                card.classList.add('played')
+                const timeOut = setTimeout(() => {
+                    card.classList.add(CARDS_HIDE_ANIMATION)
+                    card.addEventListener('animationend', function() {
+                        card.classList.remove('card-rotate')
+                        card.classList.remove('main-notActive')
+                        MAIN.classList.remove('main-notActive')
+                    }, { once: true })
+                    clearTimeout(timeOut)
+                }, 1000);
+            }
+            attempts++
+            isWin()
+        } else {
+            for (const card of openCards) {
+                const timeOut = setTimeout(() => {
+                    card.classList.remove('card-rotate')
+                    card.children[1].classList.remove('cardFace-rotate')
+                    card.children[1].addEventListener('transitionend', function() {
```

Comment:
The same recommendation. Maybe it's better to move it in one place and it once in order to prevent doing it on each card check